sys-block/partimage/partimage-0.6.9-r3.ebuild

```diff
@@ -0,0 +1,151 @@
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~sparc ~x86"
+IUSE="nls nologin pam ssl static"
+
+COMMON_DEPEND="
+	acct-group/partimag
+	acct-user/partimag
+"
+LIBS_DEPEND="
+	app-arch/bzip2
+	>=dev-libs/newt-0.52
+	>=sys-libs/slang-2
+	sys-libs/zlib:=
+	!nologin? ( virtual/libcrypt:= )
+	ssl? ( dev-libs/openssl:0= )
+"
+PAM_DEPEND="!static? ( pam? ( sys-libs/pam ) )"
+RDEPEND="
+	${COMMON_DEPEND}
+	${PAM_DEPEND}
+	!static? ( ${LIBS_DEPEND} )
+"
+DEPEND="
+	${PAM_DEPEND}
+	${LIBS_DEPEND}
+"
+BDEPEND="
+	${COMMON_DEPEND}
+	nls? ( sys-devel/gettext )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.6.9-zlib-1.2.5.2-r1.patch #405323
+	"${FILESDIR}"/${PN}-0.6.9-minor-typo.patch #580290
+	"${FILESDIR}"/${PN}-0.6.9-openssl-1.1-compatibility.patch
+	"${FILESDIR}"/${PN}-0.6.9-missing-includes.patch
+	"${FILESDIR}"/${PN}-0.6.9-clang.patch
+)
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	# XXX: Do we still need these?
+	filter-flags -fno-exceptions
+	use ppc && append-flags -fsigned-char
+
+	local myeconfargs=(
+		$(usex pam "$(usex !static '--enable-pam' '--disable-pam')" "--disable-pam")
+		$(use_enable nls)
+		$(usex nologin '--disable-login' '')
+		$(use_enable ssl)
+		$(use_enable static all-static)
+		--with-log-dir="${EPREFIX}"/var/log/partimage
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	default
+
+	keepdir /var/lib/partimage
+	keepdir /var/log/partimage
+
+	newinitd "${FILESDIR}"/partimaged.init.2 partimaged
+	newconfd "${FILESDIR}"/partimaged.conf partimaged
+
+	if use pam; then
+		newpamd "${FILESDIR}"/partimaged.pam.2 partimaged
+	fi
+
+	if use ssl; then
+		insinto /etc/partimaged
+		doins "${FILESDIR}"/servercert.cnf
+	fi
+
+	fowners partimag:root /etc/partimaged/partimagedusers
+}
+
+confdir=${EROOT}/etc/partimaged
+privkey=${confdir}/partimaged.key
+cnf=${confdir}/servercert.cnf
+csr=${confdir}/partimaged.csr
+cert=${confdir}/partimaged.cert
+
+pkg_config() {
+	if use ssl; then
+		ewarn "Please customize /etc/partimaged/servercert.cnf before you continue!"
+		ewarn "Press Ctrl-C to break now for it, or press enter to continue."
+		read
+		if [ ! -f ${privkey} ]; then
+			einfo "Generating unencrypted private key: ${privkey}"
+			openssl genrsa -out ${privkey} 2048 || die
+		else
+			einfo "Private key already exists: ${privkey}"
+		fi
+		if [ ! -f ${csr} ]; then
+			einfo "Generating certificate request: ${csr}"
+			openssl req -new -x509 -outform PEM -out ${csr} -key ${privkey} -config ${cnf} || die
+		else
+			einfo "Certificate request already exists: ${csr}"
+		fi
+		if [ ! -f ${cert} ]; then
+			einfo "Generating self-signed certificate: ${cert}"
+			openssl x509 -in ${csr} -out ${cert} -signkey ${privkey} || die
+		else
+			einfo "Self-signed certifcate already exists: ${cert}"
+		fi
+		einfo "Setting permissions"
+		partimagesslperms || die
+		einfo "Done"
+	else
+		einfo "SSL is disabled, not building certificates"
+	fi
+}
+
+partimagesslperms() {
+	local ret=0
+	fperms 600 ${privkey} 2>/dev/null
+	ret=$((${ret}+$?))
```

Comment:
Personally, I would replace this weird exit status arithmetic with `|| die` after each fperms call.

Reply:
Sounds good. Changed. Since it was originally so definied, I haven't changed it :-) It yes, it looks that way better.